test/Reducer.modifyStack.test.js

```diff
@@ -0,0 +1,378 @@
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService', index: 1 },
+        ],
+      });
+
+      expect(latestState.index).equal(2);
+      expect(latestState.children[0].sceneKey).equal('sideMenu');
+      expect(latestState.children[1].sceneKey).equal('termsOfService');
+      expect(latestState.children[2].sceneKey).equal('privacyPolicy');
+
+      currentScene = getCurrent(latestState);
+      expect(currentScene.sceneKey).equal('privacyPolicy');
+      expect(currentScene.type).equal(ActionConst.PUSH);
+    });
+
+    it('with beforeSceneKey to root stack', () => {
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService', beforeSceneKey: 'privacyPolicy' },
+        ],
+      });
+
+      expect(latestState.index).equal(2);
+      expect(latestState.children[0].sceneKey).equal('sideMenu');
+      expect(latestState.children[1].sceneKey).equal('termsOfService');
+      expect(latestState.children[2].sceneKey).equal('privacyPolicy');
+
+      currentScene = getCurrent(latestState);
+      expect(currentScene.sceneKey).equal('privacyPolicy');
+      expect(currentScene.type).equal(ActionConst.PUSH);
+    });
+
+    it('2 to root stack', () => {
+
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService' },
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'privacyPolicy', index: 1 },
+        ],
+      });
+
+      expect(latestState.index).equal(2);
+      expect(latestState.children[0].sceneKey).equal('termsOfService');
+      expect(latestState.children[1].sceneKey).equal('privacyPolicy');
+      expect(latestState.children[2].sceneKey).equal('sideMenu');
+      currentScene = getCurrent(latestState);
+      expect(currentScene.sceneKey).equal('conversations');
+      expect(currentScene.type).equal(ActionConst.PUSH);
+    });
+
+    it('cannot change current scene', () => {
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      const action = {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService', index: 2 },
+        ],
+      };
+      expect(() => reducer(latestState, action)).to.throw(/You are not allowed change current scene/);
```

Comment:
`line 289, col 1, Error - Line 289 exceeds the maximum line length of 100. (max-len)`